chevah/compat/nt_filesystem.py

```diff
@@ -978,3 +999,69 @@ def hasGroup(self, segments, group):
                 if group_sid == sid:
                     return True
         return False
+
+    def openFile(self, segments, flags, mode):
+        """
+        See `ILocalFilesystem`.
+
+        `mode` is ignored on Windows.
+        """
+        path = self.getRealPathFromSegments(segments, include_virtual=False)
+        path_encoded = self.getEncodedPath(path)
+
+        self._requireFile(segments)
+        with self._convertToOSError(path), self._impersonateUser():
+            if (
+                flags & self.OPEN_READ_ONLY == self.OPEN_READ_ONLY
+                and flags & self.OPEN_WRITE_ONLY != self.OPEN_WRITE_ONLY
+                and flags & self.OPEN_READ_WRITE != self.OPEN_READ_WRITE
+                    ):
+                # For read only mode, we use our custom code to open without
+                # a lock.
+                return self._fdRead(path_encoded)
+
+            return os.open(path_encoded, flags, mode)
+
+    def openFileForReading(self, segments):
+        """
+        See `ILocalFilesystem`.
+        """
+        path = self.getRealPathFromSegments(segments, include_virtual=False)
+        path_encoded = self.getEncodedPath(path)
+
+        self._requireFile(segments)
+        with self._convertToOSError(path), self._impersonateUser():
+            fd = self._fdRead(path_encoded)
+            return os.fdopen(fd, 'rb')
+
+    def _fdRead(self, path):
+        """
+        Do the low-level Windows file open.
+
+        Returns a file descriptor.
+        """
+        desired_access = win32file.GENERIC_READ
+        share_mode = (
+            win32file.FILE_SHARE_DELETE |
+            win32file.FILE_SHARE_WRITE |
```

Comment:
@danuker in previous version I was missing FILE_SHARE_WRITE ... so it was failing to open a file that was already opened for writing.

This should get a behaviour closer to what python os.open does.

I tried to do a quick search for python implementation of `open` but I could not find it